release/models/interfaces/openconfig-if-ip.yang

```diff
@@ -957,11 +964,41 @@ module openconfig-if-ip {
     }
 
     leaf suppress {
+      status deprecated;
       type boolean;
       default false;
       description
-        "When set to true, router advertisement neighbor discovery
-        messages are not transmitted on this interface.";
+        "When set to true, router advertisement neighbor discovery messages
+        are not transmitted on this interface.  Responses to router
+        solicitation are transmitted.";
```

Comment:
To me, this feels like a change in the existing behavior; my interpretation of the previous description is that all RAs are suppressed (solicited and unsolicited). Is this the intention?

Reply:
One of the motivations for this PR is that the existing description of behavior is a bit vague.  My interpretation of the [RFC4861](https://datatracker.ietf.org/doc/html/rfc4861#section-4 ) is that there are four separate messages in ipv6 neighbor discovery.     I think one can interpret the [current](https://github.com/openconfig/public/blob/e2799c0a9f2d9c977e404d8703971d766f64a23a/release/models/interfaces/openconfig-if-ip.yang#L959-L965) description of 'suppress'  strictly to mean never send any section 4.2 messages, would would include both solicited and unsolicited router advertisement messages as you are suggesting.   

I have also heard interpretation that the current suppress leaf should apply only the unsolicited advertisement and that responses to router solicitation are supported.  I think this is based on one interpreting a response to not be an 'advertisement', even though I think one could say this is contrary to the RFC definition.  

Debating is maybe not very useful though as the current leaf cannot meet operational need for options to suppress solicited  and unsolicited separately.

Reply:
> Debating is maybe not very useful though as the current leaf cannot meet operational need for options to suppress solicited and unsolicited separately.

My comment was more in the context of preserving backwards-compatibility for the leaf: if we clarifying the intended behavior, it seems that some of the implementations will become incompatible with the new definition

Reply:
Ok, point well taken!  I have changed the PR to keep the original description for the  suppress leaf.  If there are differences in interpretation of this deprecated leaf it's best resolved by moving to the new model.